Add: Pickup Stamina Booster

JPGalovic opened the pull request:
Stamina Booster boosts stamina by set amount when collected.
default set to 50

Commit:
Add: Pickup Stamina Booster

Stamina Booster boosts stamina by set amount when collected.
default set to 50

JPGalovic commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Pickup - Stamina Boost](https://trello.com/c/j5g3kJRn/16-pickup-stamina-boost)

Commit:
Fix: Player Movement Bug

Stamina move command should retern if player can move, not if crashed.

xxfast commented on Assets/Scripts/StaminaBar.cs:
```diff
@@ -46,26 +42,46 @@ public void recover()
-        if (stamina > 0)
-            stamina -= attackDrain;
+    // Changes stamina by delta
+    public void delta(int delta)
```
Function name isn't descriptive, maybe rename to `ChangeStaminaBy(int amound)`

JPGalovic replied:
I beg to differ, delta means change or difference, its a composite function to be used in many places. also stament above describes clearly,

JPGalovic replied:
also delta is a standard method name
I've used it heaps in many games i've developed.

xxfast commented on Assets/Scripts/StaminaBar.cs:
```diff
@@ -46,26 +42,46 @@ public void recover()
+        // Alter stamina
+        stamina += delta;
+        if (debug)
+            Debug.Log(stamina);
```
Please remove the debug messages before making a pull request. Its okay for play testing but its going to clog the console for everyone else

xxfast commented on Assets/Scripts/StaminaBar.cs:
```diff
@@ -24,20 +27,13 @@ public class StaminaBar : MonoBehaviour {
+    // Returns if player can move;
     public bool move()
     {
-        Debug.Log(stamina);
```
Please remove the debug messages before making a pull request. Its okay for play testing but its going to clog the console for everyone else

xxfast commented on Assets/Scripts/StaminaBar.cs:
```diff
@@ -24,20 +27,13 @@ public class StaminaBar : MonoBehaviour {
-            return false;
-        }
+        if (debug)
+            Debug.Log("Standard Restore: ");
```
Please remove the debug messages before making a pull request. Its okay for play testing but its going to clog the console for everyone else

JPGalovic replied:
please look at code, all debug messages are and should be put on flaggs, and no I cant remove them, they are needed for aditional development.

JPGalovic replied:
you turn em on and off in the prefab settings.

xxfast replied:
Nah, james. Its going to create a dump of logs on everyone else' terminals.

JPGalovic replied:
neither am i going to keep arguing with you or remove them, the debug elements need to remain in, you can turn on or off the display in the prefab.

xxfast commented on Assets/Scripts/StaminaBar.cs:
```diff
@@ -24,20 +27,13 @@ public class StaminaBar : MonoBehaviour {
     // Move is called each step of the players movement
-    // Returns state of stamina bar (crash or not);
+    // Returns if player can move;
     public bool move()
```
This function is not be working as intended, I tried to play test and the player is never running out of stamina at all. + player can't pick up the stamina pickup [https://media.giphy.com/media/oIWJ7oDVFlP6U/giphy.gif
![oiwj7odvflp6u](https://cloud.githubusercontent.com/assets/13775137/25007274/343493c0-20a3-11e7-96e2-122b3d8ad6ba.gif)
](url)

JPGalovic replied:
please check card, player is not picking up stamina item, that's a different card. review as intended not as you think.

JPGalovic replied:
also about player not loosing stamina, check your move drain setting, its on 0.

xxfast replied:
The card simply says 'Pickup - Stamina Boost', also the pull is named `Add: Pickup Stamina Booster` and the branch is named 'stamina-pickup'. So you're saying It **shouldn't** pick up?

JPGalovic replied:
please check the cars when you review, this task is about putting in the pickup called stamina boost, when it is picked up it will boost the players stamina, the player having the ability to collect the pickups is a differnt task blocked by this one.

xxfast commented on testBuild_Data/output_log.txt:
```diff
@@ -15,7 +15,7 @@ Platform assembly: B:\ProgramScratch\Swipe Warriors\SwipeWarriors\testBuild_Data
 Loading B:\ProgramScratch\Swipe Warriors\SwipeWarriors\testBuild_Data\Managed\UnityEngine.UI.dll into Unity Child Domain
 Platform assembly: B:\ProgramScratch\Swipe Warriors\SwipeWarriors\testBuild_Data\Managed\UnityEngine.Networking.dll (this message is harmless)
 Loading B:\ProgramScratch\Swipe Warriors\SwipeWarriors\testBuild_Data\Managed\UnityEngine.Networking.dll into Unity Child Domain
-- Completed reload, in  0.100 seconds
```
Hmm, this testBuild_Data directory shouldn't be here, because I've written the .gitignore to ignore the builds. Can you try manually deleting the directory before pull requesting? don't worry it wouldnt break the game

JPGalovic replied:
did you put the directory in git ignor or just the exe?

xxfast replied:
.gitignore is in the root directory

xxfast commented on Assets/Scripts/PickupStaminaBoost.cs:
```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PickupStaminaBoost : MonoBehaviour {
```
Better design would be to create a `PickUpBoost` base class which implements `Pickup` and `ApplyEffect` behavior and then make `PickupStaminaBoost` inherit that

JPGalovic replied:
I disagree, this is the behaviour for the particular pickup. the object itself isnt responsible for been picked up, the player is, the object simply makes its alterations. despite what you may think I have a design in mind

Commit:
Fix: Alter Delta method name

JPGalovic committed:
Cleanup: Cleaned debug on stamina output

JPGalovic committed:
Fix: Minor Errors

Fixed come minor errors in script, namly stamina not draining on attack,
script not changing correctly

xxfast committed:
Merge branch 'sprint' into stamina-pickup
